frontend/src/app/patients/AddPatient.tsx

```diff
@@ -126,12 +151,81 @@ const AddPatient = () => {
     AddPatientParams
   >(ADD_PATIENT);
 
+  const [identifyingData, setIdentifyingData] = useState<
+    Nullable<IdentifyingData>
+  >({
+    firstName: null,
+    lastName: null,
+    zipCode: null,
+    birthDate: null,
+    facilityId: null,
+  });
+  const [preventModal, setPreventModal] = useState<boolean>(false);
+
+  const [getPatientExists, { data }] = useLazyQuery(PATIENT_EXISTS, {
+    variables: {
+      ...identifyingData,
+      birthDate: moment(identifyingData.birthDate).format(
+        "YYYY-MM-DD"
+      ) as ISODate,
+    },
+  });
+
+  const facilities = useSelector<RootState, Facility[]>(
+    (state) => state.facilities
+  );
+
+  const organization = useSelector<RootState, { name: string }>(
+    (state) => state.organization
+  );
+
+  const history = useHistory();
+
+  const onBlur = ({
+    firstName,
+    lastName,
+    zipCode,
+    birthDate,
+    facilityId,
+  }: Nullable<PersonFormData>) => {
+    if (
+      firstName !== identifyingData.firstName ||
+      lastName !== identifyingData.lastName ||
+      zipCode !== identifyingData.zipCode ||
+      !moment(birthDate).isSame(identifyingData.birthDate) ||
+      facilityId !== identifyingData.facilityId
+    ) {
+      setIdentifyingData({
+        firstName,
+        lastName,
+        zipCode,
+        birthDate: moment(birthDate),
+        facilityId,
+      });
+    }
+  };
+
+  useEffect(() => {
```

Comment:
I feel like the useEffect is unnecessary, everything can be handled inside the onBlur hook

Reply:
I believe this is true but it's tempting to leave as-is under the "if it ain't broke, don't fix it" clause - especially since this pattern is duplicated wholesale in the self-registration form: https://github.com/CDCgov/prime-simplereport/blob/569dd186a0d2fe86d937fccbdfa5af5bc59b8983/frontend/src/patientApp/selfRegistration/SelfRegistrationForm.tsx#L56-L73

If you feel somewhat strongly that we should make this change here I'm glad to do it. Otherwise, we may be able to unify & simplify this approach across all workflows in a future refactor.

Reply:
yeah sounds good! I am totally down to leave it for a future refactor.